### PR TITLE
Add docs, setup more linters.

### PR DESCRIPTION
**Pull request by olomix**

### CODE REVIEW

File: claim.go [hunk 1]
```diff
@@ -44,10 +44,21 @@ Value:
...
 var ErrDataOverflow = errors.New("data does not fits SNARK size")
+
+// ErrIncorrectIDPosition means that passed position is not one of predefined:
+// IDPositionIndex or IDPositionValue
 var ErrIncorrectIDPosition = errors.New("incorrect ID position")
+
+// ErrNoID returns when ID not found in the Claim.
 var ErrNoID = errors.New("ID is not set")
 
+// ErrSlotOverflow means some DataSlow overflows Q Field. And wraps the name
```
Comment (vmidyllic) [inline claim.go:60]:
```suggestion
// ErrSlotOverflow means some DataSlot overflows Q Field. And wraps the name
```

File: claim.go [hunk 2]
```diff
@@ -113,6 +137,11 @@ type Claim struct {
 	value [4]DataSlot
 }
 
+// Subject for the time being describes the location of ID (in index or value
```
Comment (vmidyllic) [inline claim.go:140]:
```suggestion
// Subject, for the time being, describes the location of ID (in index or value
```

File: claim.go [hunk 3]
```diff
@@ -139,36 +168,42 @@ const (
 	flagUpdatableBitIdx  = 4
 )
 
+// Option provides ability to set different Claim's fields on construction
```
Comment (vmidyllic) [inline claim.go:171]:
```suggestion
// Option provides the ability to set different Claim's fields on construction
```

File: claim.go [hunk 4]
```diff
@@ -458,6 +535,8 @@ func setSlotInt(slot *DataSlot, value *big.Int, slotName SlotName) error {
 	return err
 }
 
+// TreeEntry creates new merkletree.Entry from claim. Following changes to
+// claim does not changes returned merkletree.Entry.
```
Comment (vmidyllic) [inline claim.go:539]:
```suggestion
// TreeEntry creates new merkletree.Entry from the claim. Following changes to
// claim does not change returned merkletree.Entry.
```

File: doc.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/*
+Package core contains a low level API to create and manipulate claims.
+
+Claims are creating using constructor function core.NewClaim.
+It accepts core.SchemaHash and a number of core.Option functions.
```
Comment (vmidyllic) [inline doc.go:5]:
```suggestion
Claims are created using the constructor function core.NewClaim.
It accepts core.SchemaHash and a number of core.Option functions.
```

**Review by vmidyllic — CHANGES_REQUESTED** [review]

File: claim.go (hunk 2 above)
Comment (olomix) [reply claim.go:140]: I believe this would break linter :(

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**